### PR TITLE
Add UN region codes and names

### LINKED ISSUE #20: New Zealand
Hi,

It looks like the Perl regex to match the Aland Islands also capture New Zealand, despite the `!new` instruction:

```
> countrycode("New Zealand", origin = "country.name", destination = "country.name", warn = T)
[1] "NEW ZEALAND"
Warning message:
 In countrycode("New Zealand", origin = "country.name", destination = "country.name",  :
  Some strings were matched more than once: New Zealand,ALAND ISLANDS,NEW ZEALAND
```

Any idea how the regex might be modified to effectively match only New Zealand?

### LINKED ISSUE #206: add UN region codes
Might be useful to check or enrich the package?

https://github.com/lukes/ISO-3166-Countries-with-Regional-Codes

> ISO 3166-1 country lists merged with their UN Geoscheme regional codes in ready-to-use JSON, XML, CSV data sets

### PR DESCRIPTION
**Pull request by NilsEnevoldsen**
I erred on the side of inclusion, so this adds:

1. un.region.name
2. un.region.code
3. un.subregion.name
4. un.subregion.code
5. un.intermediateregion.name
6. un.intermediateregion.code

Fixes #206

### CODE REVIEW

**Comment by vincentarelbundock** [comment]
Agree that this is useful.

Should we also rename the current `region` -> `wb.region` and `region23` -> `wb.region23` ?

**Comment by NilsEnevoldsen** [comment]
That would be a breaking change, even though they would be better names.

**Comment by NilsEnevoldsen** [comment]
What do I do to trigger a roxygen2 build? I see that codelist.Rd hasn't picked up unicode.symbol yet…

**Comment by vincentarelbundock** [comment]
`devtools::document()` should work

**Comment by NilsEnevoldsen** [comment]
Great.

**Comment by vincentarelbundock** [comment]
ready to merge?

**Comment by NilsEnevoldsen** [comment]
I was, but I'm having second thoughts about naming. What about

1. un.region.name
2. un.region.code
3. un.regionsub.name
4. un.regionsub.code
5. un.regionintermediate.name
6. un.regionintermediate.code

to keep them all together?

**Comment by vincentarelbundock** [comment]
Yeah, I like this better.

Does @cjyetman have a view on backward compatibility and the question I asked above about renaming the wb regions?

File: R/codelist.R [hunk 1]
```diff
@@ -50,12 +50,18 @@
 #'   \item p4.name: Polity IV country name
 #'   \item region: 7 Regions as defined in the World Bank Development Indicators
 #'   \item region23: 23 Regions as used to be in the World Bank Development Indicators (legacy)
+#'   \item un.intermediateregion.name: United Nations intermediate region name
+#'   \item un.intermediateregion.name: United Nations intermediate region code
```
Comment (cjyetman) [inline R/codelist.R:54]: probably should be `\item un.intermediateregion.code: United Nations intermediate region code`

File: R/codelist.R [hunk 2]
```diff
@@ -50,12 +50,18 @@
...
 #'   \item un.name.ar: United Nations Arabic country name
 #'   \item un.name.en: United Nations English country name
 #'   \item un.name.es: United Nations Spanish country name
 #'   \item un.name.fr: United Nations French country name
 #'   \item un.name.ru: United Nations Russian country name
 #'   \item un.name.zh: United Nations Chinese country name
+#'   \item un.region.name: United Nations region name
+#'   \item un.region.name: United Nations region code
```
Comment (cjyetman) [inline R/codelist.R:62]: probably should be `\item un.region.code: United Nations region code`

File: R/codelist.R (hunk 1 above)
Comment (NilsEnevoldsen) [reply R/codelist.R:54]: Oops

File: R/codelist.R [hunk 3]
```diff
@@ -50,12 +50,18 @@
...
+#'   \item un.subregion.name: United Nations sub-region name
+#'   \item un.subregion.name: United Nations sub-region code
```
Comment (cjyetman) [inline R/codelist.R:64]: probably should be `\item un.subregion.code: United Nations sub-region code`

Comment (NilsEnevoldsen) [reply R/codelist.R]: Oops

File: R/codelist.R (hunk 2 above)
Comment (NilsEnevoldsen) [reply R/codelist.R]: Oops

**Comment by NilsEnevoldsen** [comment]
@vincentarelbundock Assuming that by "this" you meant my recently proposed names, this change implements those.

**Comment by vincentarelbundock** [comment]
Thanks!

**Pull request merged**